plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -32,16 +35,25 @@ def extract_filepath(provided_regex: str, provided_string: str, provided_file: s
 def strip_subdomains(matches: list) -> list:
     for match in enumerate(matches):
         stripped_domain = tldextract.extract(match[1])
-        matches[match[0]] = ".".join(stripped_domain[1:3])
+        if not stripped_domain.suffix:
+            suffix = stripped_domain.domain
+            subdomain = stripped_domain.subdomain
+            if subdomain and suffix:
+                subdomains = subdomain.split(".")
+                matches[match[0]] = f"{subdomains[len(subdomains) - 1]}.{suffix}"
+        else:
+            matches[match[0]] = ".".join(stripped_domain[1:3])
```

Comment:
Can you add some comments explaining the process here? It's a little difficult to follow this logic.